compiler/tools/fpp-to-json/test/python/locationMapValidator.py

```diff
@@ -0,0 +1,50 @@
+import json
+import sys
+import re
+
+def collect_ids(json_data):
+    ids = []
+
+    if isinstance(json_data, list):
+        for item in json_data:
+            ids.extend(collect_ids(item))
+
+    elif isinstance(json_data, dict):
+        for key, value in json_data.items():
+            if key == 'id':
+                # Check if the value is an integer
+                if isinstance(value, int):
+                    ids.append(value)
+                # Check if the value is a string and matches the format 'id : number'
+                elif isinstance(value, str) and re.match(r"id\s*:\s*\d+$", value):
+                    ids.append(int(value.split(':')[1].strip()))
```

Comment:
This code seems to assume that we can have a string "id: 5" in the output that actually represents an object with a field id with value 5. This should never happen in our JSON output. Can you confirm this never happens? If so, let's remove this code. If not, let's fix it in the JSON encoder.